src/web-api.js

```diff
@@ -375,6 +375,75 @@ class WebApi {
       .get_job_status(partner_params.user_id, partner_params.job_id, options);
   }
 
+	get_web_token(requestParams) {
+		return new Promise((resolve, reject) => {
+			if (!requestParams) {
+				reject( new Error('Please ensure that you send through request params') );
+			}
+
+			if (typeof requestParams !== 'object') {
+				reject( new Error('Request params needs to be an object') );
+			}
+
+			['user_id', 'job_id', 'product'].forEach(requiredParam => {
```

Comment:
You should check that either the class was instantiated with a callback url or that one was passed in request params also.

Reply:
gotcha. thanks.

Reply:
should that be repeated here though? considering it is done in the lambda?

Reply:
doing so here will mean we pass an `options` object, with a `callback` entry. 

at least to follow convention

Reply:
yes it should be here too. There is no point in allowing requests you know will fail. It only slows down the feedback cycle for those integrating. We validate in the lambda for requests that might not be coming from our library.

Reply:
added the check for `callback_url` or `this.default_callback` above this